datalad/support/gitrepo.py

```diff
@@ -3109,12 +3154,14 @@ def _diffstatus_get_state_props(self, f, from_state, to_state,
         Parameters
         ----------
         f : Path
-        from_state : dict
-        to_state : dict
+        from_state : dict or None
+        to_state : dict or None
         against_commit : bool
           Flag whether `to_state` reflects a commit or the worktree.
-        modified_in_worktree : bool
-          Flag whether a worktree modification is reported. This is ignored
+        modified_in_worktree : False or str
```

Comment:
```suggestion
        modified_in_worktree : False or str or None
```

Reply:
I don't know what makes you suggest this change. If it is based on a condition you saw happen in the code, that condition could be a bug. The only time this helper is called ensures that `None` is not passed as an argument.